### ISSUE TITLE
SR OS crash when no keys are found

### ISSUE THREAD

**Issue by hellt** [srl-labs/containerlab#1722]
```
INFO[0016] Waiting for clab-jgc-lab-sros3 to be ready. This may take a while. Monitor boot log with `sudo docker logs -f clab-jgc-lab-sros3`
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x24d73e2]
 
goroutine 59 [running]:
github.com/srl-labs/containerlab/nodes/vr_sros.(*vrSROS).applyPartialConfig(0xc000a58000, {0x352c0a8, 0xc000560e10}, {0xc0003ad020, 0xb}, {0x2f0dbde, 0xa}, {0x2f0237b, 0x5}, {0x2f0237b, ...}, ...)
                github.com/srl-labs/containerlab/nodes/vr_sros/vr-sros.go:285 +0x862
github.com/srl-labs/containerlab/nodes/vr_sros.(*vrSROS).configureSSHPublicKeys(0xc000a58000, {0x352c0a8, 0xc000560e10}, {0xc0005d1ed0?, 0x40db48?}, {0xc000a4c0f0?, 0x0?}, {0xc000560e10?, 0xc0003b81e0?}, {0x2f0237b, ...}, ...)
                github.com/srl-labs/containerlab/nodes/vr_sros/sshKey.go:80 +0x425
github.com/srl-labs/containerlab/nodes/vr_sros.(*vrSROS).PostDeploy(0xc000a58000, {0x352c0a8, 0xc000560e10}, 0x0?)
                github.com/srl-labs/containerlab/nodes/vr_sros/vr-sros.go:137 +0x365
github.com/srl-labs/containerlab/cmd.deployFn.func1({0x354f3a8, 0xc000a58000}, 0xc000a5e370?)
```

**Comment by steiler** [comment]
Hmmm, so in this case we must have found some keys. otherwise this would have prevented us from going into vr_sros/sshKey.go:80
https://github.com/srl-labs/containerlab/blob/4e4748c7b2099f520fdac8f72e99f7f8ec449e27/nodes/vr_sros/vr-sros.go#L136
I could imagine that there is still a different key type from those that we have listed for mapping and that the mapping results in two empty list.
https://github.com/srl-labs/containerlab/blob/4e4748c7b2099f520fdac8f72e99f7f8ec449e27/nodes/vr_sros/sshKey.go#L58-L64
This would lead to a template renderign, that would simply yield an empty config snippet, that might cause issues.
I'll double check these assumptions.

**Comment by hellt** [comment]
Yes, this might have been ed25519 key that is the default key type in ssh keygen now

**Comment by steiler** [comment]
alright, I'll try that... would be an easy fix if its just that.

**Comment by steiler** [comment]
I can't reproduce this issue. Even when only unsupported keys (ED25519) are present we push the following config line:
https://github.com/srl-labs/containerlab/blob/4e4748c7b2099f520fdac8f72e99f7f8ec449e27/nodes/vr_sros/ssh_keys.go.tpl#L3-L5
So no need to panic here.

**Comment by steiler** [comment]
Ok, got it now, I guess.
Looking at the line that caused the failure:
https://github.com/srl-labs/containerlab/blob/4e4748c7b2099f520fdac8f72e99f7f8ec449e27/nodes/vr_sros/vr-sros.go#L285
Taking into account we received a nil pointer, the `mr` var is nil. So we cannot access `mr`s fields.
scraply definetly returns nils for mr see:
https://github.com/scrapli/scrapligo/blob/9f78e4bbd4d024853d61859ea6b65d7da52e9111/driver/network/sendconfigs.go#L17-L38
So if we need to print the `mr.Failure` field, we should perform an `!= nil` check upfront

**Comment by hellt** [comment]
fixed in #1723
